Fix data security consideration in how ASM works (#17332)
DataDog/documentation: content/en/security/application_security/how-appsec-works.md

## Data privacy

By default, ASM collects information from suspicious requests to help you understand why the request was flagged as suspicious. Before sending the data, ASM scans it for patterns and keywords that indicate that the data is sensitive. If the data is deemed sensitive, it is replaced with a <redacted> flag, so you observe that although the request was suspicious, the request data could not be collected because of data security concerns.

> The following are examples of data that are flagged as sensitive by default:

Reviewer: ```suggestion
Here are some examples of data that is flagged as sensitive by default:
```
